inc/search.class.php

```diff
@@ -3406,11 +3406,12 @@ static function addSelect($itemtype, $ID, $meta = 0, $meta_type = 0) {
          $complexjoin = self::computeComplexJoinID($searchopt[$ID]['joinparams']);
       }
 
+      $fk = getForeignKeyFieldForTable($table);
       if (((($table != getTableForItemType($itemtype))
             && (!isset($CFG_GLPI["union_search_type"][$itemtype])
                 || ($CFG_GLPI["union_search_type"][$itemtype] != $table)))
            || !empty($complexjoin))
-          && ($searchopt[$ID]["linkfield"] != getForeignKeyFieldForTable($table))) {
+          && ($searchopt[$ID]["linkfield"] != $fk && strpos($searchopt[$ID]["linkfield"], $fk . '_') !== 0)) {
```

Comment:
Maybe it could even be simplier with following code, unless I miss something.
```suggestion
          && getTableNameForForeignKeyField($searchopt[$ID]["linkfield"]) != $table) {
```

Reply:
Seems OK to me